Modules/_pickle.c

```diff
@@ -1679,15 +1677,16 @@ whichmodule(PyObject *global, PyObject *dotted_path)
     assert(module_name == NULL);
 
     /* Fallback on walking sys.modules */
-    modules_dict = _PySys_GetObjectId(&PyId_modules);
-    if (modules_dict == NULL) {
+    modules = _PySys_GetObjectId(&PyId_modules);
+    if (modules == NULL) {
         PyErr_SetString(PyExc_RuntimeError, "unable to get sys.modules");
         return NULL;
     }
 
-    i = 0;
-    while (PyDict_Next(modules_dict, &i, &module_name, &module)) {
+    PyObject *iterator = PyObject_GetIter(modules);
```

Comment:
Where does this iterator get cleaned up?

Reply:
Thank you!  I would have found it eventually but you just saved me a bunch of time!